pkgs/top-level/all-packages.nix

```diff
@@ -14898,16 +14898,24 @@ with pkgs;
   julia-stable = julia_18;
   julia = julia-stable;
 
-  julia_16-bin = callPackage ../development/compilers/julia/1.6-bin.nix { };
-  julia_18-bin = callPackage ../development/compilers/julia/1.8-bin.nix { };
+  julia_16-bin =  let julia_16-bin' = callPackage ../development/compilers/julia/1.6-bin.nix { };
+                  in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_16-bin'; };
 
-  julia_18 = callPackage ../development/compilers/julia/1.8.nix { };
-  julia_19 = callPackage ../development/compilers/julia/1.9.nix { };
+  julia_18-bin = let julia_18-bin' = callPackage ../development/compilers/julia/1.8-bin.nix { };
+                 in callPackage ../development/compilers/julia/with-packages.nix { julia = julia_18-bin'; };
+
+  julia_18 = let julia_18' = callPackage ../development/compilers/julia/1.8.nix { };
```

Comment:
I think julia-bin and so on should still have the correct meta information and version set such that with-packages needs to passthrough those or is this not the common approach?